### DISCUSSION TITLE
How to debug the POST /store endpoint?

### DISCUSSION THREAD

**Discussion by ships** [oxigraph/oxigraph#584] [Q&A] [answered]
Hi friends and thanks for this fascinating work!

I am trying to explore an RDF dataset using Oxigraph to understand how to structure SPARQL queries against it. My basic flow is to start Oxigraph on Docker in an empty data directory:

`docker run --rm -v $PWD/graph_data:/data -p 7878:7878 ghcr.io/oxigraph/oxigraph --location /data serve --bind 0.0.0.0:7878`

and then POST:

`curl -vvvf -X POST -H 'Content-Type:application/n-triples' -T some_docmaps.nt http://127.0.0.1:7878/store`

I see that I get a 201 Created, even though no logs are emitted to the container's stdout; and anyway the `graph_data` directory has a few hundred mb of data in it now.

but when i submit a generic sparql query to validate:

```sparql
SELECT * WHERE {
  ?sub ?pred ?obj .
} LIMIT 10
```

I get no results!

I think I am misunderstanding Oxigraph's handling of default graphs or something, can you point me in the right direction to understand that or a way to get more verbose info about what the server is doing while it handles the POST?

**Comment by Tpt** [comment, marked as answer]
Hi! Thank you for trying Oxigraph.

Yes, the SPARQL graph store protocol has a few behaviors that are deeply disturbing for people who don't know it by hart.

What is happening is that when posting to `http://127.0.0.1:7878/store` a new named graph is created, hence the 201 Created response. The name of the name graph is in the returned `Location` header. The SPARQL query does not work because by default it only queries the default graph you need to add `?union-default-graph=true` to the URL to query the union of all graphs or execute `SELECT * WHERE { GRAPH ?g { ?s ?p ?o }} LIMIT 10`. If you want to add data to the default graph, you need to POST to `http://127.0.0.1:7878/store?default` instead.

> even though no logs are emitted

yes, that's something I should fix. I add it to my "priority" TODO.

**Reply by ships** [reply]
thank you @Tpt that resolves my issue and educates me a lot!
